### PR TITLE
Ported all listeners to middlewres

### PR DESCRIPTION
**Pull request by Nyholm**
Note that this PR should be merged to #267

### CODE REVIEW

File: lib/Buzz/Middleware/CallbackMiddleware.php [hunk 1]
```diff
@@ -37,13 +38,17 @@ public function __construct($callable)
         $this->callable = $callable;
     }
 
-    public function preSend(RequestInterface $request)
+    public function handleRequest(RequestInterface $request, callable $next)
     {
         call_user_func($this->callable, $request);
```
Comment (stof) [inline lib/Buzz/Middleware/CallbackMiddleware.php]: this does not allow altering the request anymore, as the PSR-7 one is immutable.

You should allow the callback to return a modified request and use it for the next call (if the return value is `null`, use the original one, or throw if you want to mandate returning a request)

File: lib/Buzz/Middleware/CallbackMiddleware.php [hunk 2]
```diff
@@ -37,13 +38,17 @@ public function __construct($callable)
...
+
+        return $next($request);
     }
 
-    public function postSend(RequestInterface $request, MessageInterface $response)
+    public function handleResponse(RequestInterface $request, ResponseInterface $response, callable $next)
     {
         call_user_func($this->callable, $request, $response);
```
Comment (stof) [inline lib/Buzz/Middleware/CallbackMiddleware.php]: same here. The callback cannot alter the response

File: .travis.yml [hunk 3]
```diff
@@ -42,7 +44,7 @@ before_script:
     - vendor/bin/http_test_server > /dev/null 2>&1 &
 
 script:
-    - composer validate --strict --no-check-lock
+    #- composer validate --strict --no-check-lock
```
Comment (stof) [inline .travis.yml]: why commenting this ?

Comment (Nyholm) [reply .travis.yml]: Because I was using dev-master of psr/http-client

File: lib/Buzz/Middleware/CallbackMiddleware.php (hunk 1 above)
Comment (Nyholm) [reply lib/Buzz/Middleware/CallbackMiddleware.php]: This might be a mistake. Thanks!

**Pull request merged**